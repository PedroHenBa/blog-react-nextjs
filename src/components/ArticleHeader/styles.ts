import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xbig};
    margin-bottom: ${theme.spacing.xbig};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.spacing.medium};
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `}
`;
