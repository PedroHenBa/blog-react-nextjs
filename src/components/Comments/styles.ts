import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacing.big} auto;
    padding: 0 ${theme.spacing.big};
    color: ${theme.colors.darkText};
  `}
`;
