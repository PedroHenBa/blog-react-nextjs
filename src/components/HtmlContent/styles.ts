import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.medium};

    p {
      margin: ${theme.spacing.big} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      transition: all 200ms ease-in-out;
    }

    a:hover {
      filter: brightness(70%);
    }

    code {
      font-family: monospace;
      color: ${theme.colors.secondary};
      font-size: ${theme.fonts.sizes.small};
      background: ${theme.colors.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.primary};
      padding: ${theme.spacing.medium};
      color: ${theme.colors.white};
      font-family: monospace;
      margin: ${theme.spacing.medium} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.fonts.sizes.small};
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacing.medium} 0;
    }

    .image figcaption {
      padding: ${theme.spacing.small};
      text-align: center;
      font-size: ${theme.fonts.sizes.small};
      line-height: 1.3;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacing.medium} 0;
    }

    ul,
    ol {
      margin: ${theme.spacing.medium} ${theme.spacing.xbig};
    }

    hr {
      border: none;
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      border-collapse: collapse;
      border: 1px solid #eee;
      border-bottom: 2px solid #00cccc;
      box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.1), 0px 10px 20px rgba(0, 0, 0, 0.05), 0px 20px 20px rgba(0, 0, 0, 0.05),
        0px 30px 20px rgba(0, 0, 0, 0.05);
      tr {
        &:hover {
          background: #f4f4f4;

          td {
            color: #555;
          }
        }
      }
      th,
      td {
        color: #999;
        border: 1px solid #eee;
        padding: 12px 35px;
        border-collapse: collapse;
      }
      th {
        background: #00cccc;
        color: #fff;
        text-transform: uppercase;
        &.last {
          border-right: none;
        }
      }
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkGray};
      filter: brightness(80%);
      padding-left: ${theme.spacing.medium};
      font-style: italic;
      margin: ${theme.spacing.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
