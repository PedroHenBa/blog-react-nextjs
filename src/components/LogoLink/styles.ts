import styled from 'styled-components';

export const Container = styled.a`
  display: flex;
  color: inherit;
  text-decoration: none;
  align-items: center;

  > img {
    width: 15rem;
    height: 15rem;
    border-radius: 50%;
  }
`;
