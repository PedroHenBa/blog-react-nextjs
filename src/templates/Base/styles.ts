import styled, { css } from 'styled-components';

export const Container = styled.div``;

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacing.xhuge};
    padding: 0 ${theme.spacing.big};
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacing.xhuge};
  `}
`;

export const FooterContainer = styled.div`
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
`;
