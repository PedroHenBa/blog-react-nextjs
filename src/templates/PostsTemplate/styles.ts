import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacing.xsmall} ${theme.spacing.small};
    border-radius: ${theme.spacing.xsmall};
    border: none;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    cursor: pointer;

    &:active {
      transform: translateY(-1px);
      box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
    }

    &:disabled {
      background: ${theme.colors.darkText};
      color: ${theme.colors.white};
    }
  `}
`;

export const ButtonContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
`;
