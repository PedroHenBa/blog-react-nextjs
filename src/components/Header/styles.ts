import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacing.xhuge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacing.small} - 1rem);
    }

    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeadingStyles} {
        margin: 0 0 ${theme.spacing.medium} 0;
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    flex-flow: row wrap;
    margin-left: ${theme.spacing.medium};
    max-width: 44rem;

    @media ${theme.media.lteSmall} {
      margin-left: 0;
    }
  `}
`;
