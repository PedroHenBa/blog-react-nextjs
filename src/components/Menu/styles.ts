import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export type MenuOpen = {
  isMenuOpen: boolean;
};

const styledMenuIcon = (theme: DefaultTheme, isMenuOpen: MenuOpen['isMenuOpen']) => {
  if (isMenuOpen) {
    return css`
      color: ${theme.colors.secondary};
      left: 26rem;
    `;
  } else {
    return css`
      color: ${theme.colors.white};
      left: 0rem;
    `;
  }
};

export const Container = styled.div<MenuOpen>`
  ${({ theme, isMenuOpen }) => css`
    z-index: 1;
    position: fixed;
    padding: ${theme.spacing.big};
    background: ${theme.colors.primary};
    height: 100vh;
    width: 100%;
    max-width: 32rem;
    display: flex;
    top: 0;
    flex-flow: column nowrap;
    left: ${isMenuOpen ? 0 : '-31rem'};
    overflow-y: ${isMenuOpen ? 'auto' : 'hidden'};
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      left: ${isMenuOpen ? 0 : '-32rem'};
    }
  `}
`;

export const MenuOpen = styled.a<MenuOpen>`
  ${({ theme, isMenuOpen }) => css`
    z-index: 2;
    position: fixed;
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    width: 3rem;
    height: 3rem;
    top: ${theme.spacing.medium};
    ${styledMenuIcon(theme, isMenuOpen)}
    transition: all 300ms ease-in-out;
  `}
`;

export const NavLinks = styled.nav`
  margin: auto;
  width: 100%;
`;

export const NavLogo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacing.xxbig};
      img {
        border: 0.5rem solid ${theme.colors.secondary};
      }
    }
  `}
`;
