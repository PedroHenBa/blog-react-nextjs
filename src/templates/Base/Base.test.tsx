import { BaseTemplate } from './index';
import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import props from './mock';

describe('<BaseTemplate />', () => {
  it('should render BaseTemplate component', () => {
    renderTheme(<BaseTemplate {...props} />);
    expect(screen.getByRole('img', { name: `${props.settings.blogName} - ${props.settings.blogDescription}` }));
    expect(screen.getByLabelText('Open or close menu')).toBeInTheDocument();
    expect(screen.getByText('Feito com carinho, ph')).toBeInTheDocument();
    expect(screen.getByLabelText('Go to top')).toBeInTheDocument();
  });

  it('should render base elements', () => {
    const { container } = renderTheme(<BaseTemplate {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
