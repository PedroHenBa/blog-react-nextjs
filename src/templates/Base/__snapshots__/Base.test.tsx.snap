// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseTemplate /> should render base elements 1`] = `
.c13 {
  margin-top: 6.4rem;
  padding: 0 3.2rem;
}

.c17 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c18 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
}

.c9 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c16 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c6 {
  z-index: 1;
  position: fixed;
  padding: 3.2rem;
  background: #000000;
  height: 100vh;
  width: 100%;
  max-width: 32rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  top: 0;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  left: -31rem;
  overflow-y: hidden;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c4 {
  z-index: 2;
  position: fixed;
  color: #FFFFFF;
  background: #000000;
  width: 3rem;
  height: 3rem;
  top: 2.4rem;
  color: #FFFFFF;
  left: 0rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c11 {
  margin: auto;
  width: 100%;
}

.c7 .c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c7 .c8 img {
  border: 0.5rem solid #dc143c;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c12 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c12:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  margin: 0 auto;
}

.c14 .c8 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c15 {
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-left: 2.4rem;
  max-width: 44rem;
}

.c22 {
  font-size: 2.4rem;
}

.c22 p {
  margin: 3.2rem 0;
}

.c22 a,
.c22 a:visited,
.c22 a:link {
  color: #dc143c;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c22 a:hover {
  -webkit-filter: brightness(70%);
  filter: brightness(70%);
}

.c22 img {
  max-width: 100%;
}

.c22 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c22 .image figcaption {
  padding: 1.6rem;
  text-align: center;
  font-size: 1.6rem;
  line-height: 1.3;
}

.c22 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c22 ul,
.c22 ol {
  margin: 2.4rem 4.0rem;
}

.c22 hr {
  border: none;
  border: 0.1rem solid #DDDDDD;
}

.c22 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c22 table {
  border-collapse: collapse;
  border: 1px solid #eee;
  border-bottom: 2px solid #00cccc;
  box-shadow: 0px 0px 20px rgba(0,0,0,0.1),0px 10px 20px rgba(0,0,0,0.05),0px 20px 20px rgba(0,0,0,0.05),0px 30px 20px rgba(0,0,0,0.05);
}

.c22 table tr:hover {
  background: #f4f4f4;
}

.c22 table tr:hover td {
  color: #555;
}

.c22 table th,
.c22 table td {
  color: #999;
  border: 1px solid #eee;
  padding: 12px 35px;
  border-collapse: collapse;
}

.c22 table th {
  background: #00cccc;
  color: #fff;
  text-transform: uppercase;
}

.c22 table th.last {
  border-right: none;
}

.c22 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c19 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c19 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c19 .c21 {
  font-size: 1.6rem;
}

.c20 pre[class*="language-"],
.c20 code[class*="language-"] {
  color: #d4d4d4;
  font-size: 13px;
  text-shadow: none;
  font-family: Menlo,Monaco,Consolas,"Andale Mono","Ubuntu Mono","Courier New",monospace;
  direction: ltr;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  line-height: 1.5;
  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.c20 pre[class*="language-"]::selection,
.c20 code[class*="language-"]::selection,
.c20 pre[class*="language-"] *::selection,
.c20 code[class*="language-"] *::selection {
  text-shadow: none;
  background: #75a7ca;
}

.c20 pre[class*="language-"] {
  padding: 1em;
  margin: .5em 0;
  overflow: auto;
  background: #1e1e1e;
}

.c20:not(pre) > code[class*="language-"] {
  padding: .1em .3em;
  border-radius: .3em;
  color: #db4c69;
  background: #f9f2f4;
}

.c20 .namespace {
  opacity: .7;
}

.c20 .token.doctype .token.doctype-tag {
  color: #569CD6;
}

.c20 .token.doctype .token.name {
  color: #9cdcfe;
}

.c20 .token.comment,
.c20 .token.prolog {
  color: #6a9955;
}

.c20 .token.punctuation,
.c20 .language-html .language-css .token.punctuation,
.c20 .language-html .language-javascript .token.punctuation {
  color: #d4d4d4;
}

.c20 .token.property,
.c20 .token.tag,
.c20 .token.boolean,
.c20 .token.number,
.c20 .token.constant,
.c20 .token.symbol,
.c20 .token.inserted,
.c20 .token.unit {
  color: #b5cea8;
}

.c20 .token.selector,
.c20 .token.attr-name,
.c20 .token.string,
.c20 .token.char,
.c20 .token.builtin,
.c20 .token.deleted {
  color: #ce9178;
}

.c20 .language-css .token.string.url {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c20 .token.operator,
.c20 .token.entity {
  color: #d4d4d4;
}

.c20 .token.operator.arrow {
  color: #569CD6;
}

.c20 .token.atrule {
  color: #ce9178;
}

.c20 .token.atrule .token.rule {
  color: #c586c0;
}

.c20 .token.atrule .token.url {
  color: #9cdcfe;
}

.c20 .token.atrule .token.url .token.function {
  color: #dcdcaa;
}

.c20 .token.atrule .token.url .token.punctuation {
  color: #d4d4d4;
}

.c20 .token.keyword {
  color: #569CD6;
}

.c20 .token.keyword.module,
.c20 .token.keyword.control-flow {
  color: #c586c0;
}

.c20 .token.function,
.c20 .token.function .token.maybe-class-name {
  color: #dcdcaa;
}

.c20 .token.regex {
  color: #d16969;
}

.c20 .token.important {
  color: #569cd6;
}

.c20 .token.italic {
  font-style: italic;
}

.c20 .token.constant {
  color: #9cdcfe;
}

.c20 .token.class-name,
.c20 .token.maybe-class-name {
  color: #4ec9b0;
}

.c20 .token.console {
  color: #9cdcfe;
}

.c20 .token.parameter {
  color: #9cdcfe;
}

.c20 .token.interpolation {
  color: #9cdcfe;
}

.c20 .token.punctuation.interpolation-punctuation {
  color: #569cd6;
}

.c20 .token.boolean {
  color: #569cd6;
}

.c20 .token.property,
.c20 .token.variable,
.c20 .token.imports .token.maybe-class-name,
.c20 .token.exports .token.maybe-class-name {
  color: #9cdcfe;
}

.c20 .token.selector {
  color: #d7ba7d;
}

.c20 .token.escape {
  color: #d7ba7d;
}

.c20 .token.tag {
  color: #569cd6;
}

.c20 .token.tag .token.punctuation {
  color: #808080;
}

.c20 .token.cdata {
  color: #808080;
}

.c20 .token.attr-name {
  color: #9cdcfe;
}

.c20 .token.attr-value,
.c20 .token.attr-value .token.punctuation {
  color: #ce9178;
}

.c20 .token.attr-value .token.punctuation.attr-equals {
  color: #d4d4d4;
}

.c20 .token.entity {
  color: #569cd6;
}

.c20 .token.namespace {
  color: #4ec9b0;
}

.c20 pre[class*="language-javascript"],
.c20 code[class*="language-javascript"],
.c20 pre[class*="language-jsx"],
.c20 code[class*="language-jsx"],
.c20 pre[class*="language-typescript"],
.c20 code[class*="language-typescript"],
.c20 pre[class*="language-tsx"],
.c20 code[class*="language-tsx"] {
  color: #9cdcfe;
}

.c20 pre[class*="language-css"],
.c20 code[class*="language-css"] {
  color: #ce9178;
}

.c20 pre[class*="language-html"],
.c20 code[class*="language-html"] {
  color: #d4d4d4;
}

.c20 .language-regex .token.anchor {
  color: #dcdcaa;
}

.c20 .language-html .token.punctuation {
  color: #808080;
}

.c20 pre[data-line] {
  position: relative;
}

.c20 pre[class*="language-"] > code[class*="language-"] {
  position: relative;
  z-index: 1;
}

.c20 .line-highlight {
  position: absolute;
  left: 0;
  right: 0;
  padding: inherit 0;
  margin-top: 1em;
  background: #f7ebc6;
  box-shadow: inset 5px 0 0 #f7d87c;
  z-index: 0;
  pointer-events: none;
  line-height: inherit;
  white-space: pre;
}

.c23 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 34px;
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c2:checked + span {
  background-color: #dc143c;
}

.c2:focus + span {
  box-shadow: 0 0 1px #2196f3;
}

.c2:checked + span:before {
  -webkit-transform: translateX(26px);
  -ms-transform: translateX(26px);
  -webkit-transform: translateX(26px);
  -ms-transform: translateX(26px);
  transform: translateX(26px);
}

.c3 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c3:before {
  position: absolute;
  content: '';
  height: 26px;
  width: 26px;
  left: 4px;
  bottom: 4px;
  background-color: white;
  -webkit-transition: 0.4s;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c3 {
  border-radius: 34px;
}

.c3:before {
  border-radius: 50%;
}

@media (max-width:768px) {
  .c6 {
    left: -32rem;
  }
}

@media (max-width:360px) {
  .c14 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
  }

  .c14 > .c8 {
    margin: 0 0 2.4rem 0;
  }
}

@media (max-width:360px) {
  .c15 {
    margin-left: 0;
  }
}

@media (max-width:768px) {
  .c22 {
    font-size: 2rem;
  }

  .c22 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media print {
  .c20 pre[class*="language-"],
  .c20 code[class*="language-"] {
    text-shadow: none;
  }
}

<div
  class="styles__Container-sc-550pzn-0"
>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      <input
        class="c2"
        type="checkbox"
        value="Dark mode active"
      />
      <span
        class="c3"
      />
    </label>
  </div>
  <a
    aria-label="Open or close menu"
    class="c4"
    href="#"
    title="Open or close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </a>
  <div
    aria-hidden="true"
    class="c6"
  >
    <div
      class="c7"
    >
      <h1
        class="c8 c9"
      >
        <a
          class="c10"
          href="/"
          target="_self"
        >
          <img
            alt="Bloguinho"
            src="https://res.cloudinary.com/djve6xgfe/image/upload/v1627833918/na_14eb8ec6d4.jpg"
          />
        </a>
      </h1>
    </div>
    <nav
      class="c11"
    >
      <a
        class="c12"
        href="link1"
        target="_self"
      >
        text1
      </a>
      <a
        class="c12"
        href="link2"
        target="_self"
      >
        link2
      </a>
    </nav>
  </div>
  <div
    class="c13"
  >
    <header
      class="c14"
    >
      <h1
        class="c8 c9"
      >
        <a
          class="c10"
          href="/"
          target="_self"
        >
          <img
            alt="Bloguinho - am tincidunt ipsum et imperdiet tincidunt. Fusce eget elementum mi. Nulla pharetra aliquet turpis sit amet elementum. In placerat enim ut nequ"
            src="https://res.cloudinary.com/djve6xgfe/image/upload/v1627833918/na_14eb8ec6d4.jpg"
          />
        </a>
      </h1>
      <div
        class="c15"
      >
        <h2
          class="c8 c16"
        >
          Bloguinho
        </h2>
        <p>
          am tincidunt ipsum et imperdiet tincidunt. Fusce eget elementum mi. Nulla pharetra aliquet turpis sit amet elementum. In placerat enim ut nequ
        </p>
      </div>
    </header>
  </div>
  <div
    class="c17"
  >
    Salve
  </div>
  <div
    class="c18"
  >
    <footer
      class="c19"
    >
      <div
        class="c20"
      >
        <div
          class="c21 c22"
        >
          <p>
            Feito com carinho, ph
          </p>
        </div>
        ;
      </div>
    </footer>
  </div>
  <a
    aria-label="Go to top"
    class="c23"
    href="#"
    title="Go to top"
  >
    <svg
      aria-hidden="true"
      class="c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
